### PR TITLE
Collect slow and frozen frames for spans

### PR DESCRIPTION
**Pull request by markushi**
By utilizing our existing `SentryFrameMetricsCollector` via a new performance collector:`SpanFrameMetricsCollector`

## :scroll: Description
Fixes https://github.com/getsentry/sentry-java/issues/3106

Based on https://github.com/getsentry/sentry-java/pull/3102, which needs to merged first.

If perf-v2 is enabled, the `ActivityFramesTracker` will no-op and instead `SpanFrameMetricsCollector` will start collecting the data on a span level (including the top level transaction `ISpan`).

## :bulb: Motivation and Context
Top level epic: https://github.com/getsentry/team-mobile/issues/156

## :green_heart: How did you test it?
Added unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    normalFrameCount = 0;
+
+    slowFrameCount = 0;
+    slowFrameDuration = 0;
+
+    frozenFrameCount = 0;
+    frozenFrameDuration = 0;
+  }
+
+  public FrameMetrics duplicate() {
```
Comment (romtsn) [inline sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java:80]: nit: we usually call it `clone()` in other places

File: sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    return new FrameMetrics(
+        normalFrameCount, slowFrameCount, slowFrameDuration, frozenFrameCount, frozenFrameDuration);
+  }
+
+  /**
+   * @param other the other frame metrics to compare to, usually the older one
+   * @return the difference between two frame metrics (this minus other)
+   */
+  public FrameMetrics diffTo(FrameMetrics other) {
```
Comment (romtsn) [inline sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java:89]:
```suggestion
  public FrameMetrics diffTo(final @NotNull FrameMetrics other) {
```

File: sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+public class SpanFrameMetricsCollector
+    implements IPerformanceContinuousCollector,
+        SentryFrameMetricsCollector.FrameMetricsCollectorListener {
+  private @NotNull final Object lock = new Object();
+  private @Nullable final SentryFrameMetricsCollector frameMetricsCollector;
+  private @Nullable volatile String listenerId;
+  private @NotNull final Map<SpanId, FrameMetrics> metricsAtSpanStart;
+
+  private @NotNull final FrameMetrics currentFrameMetrics;
+  float lastRefreshRate = 0;
```
Comment (romtsn) [inline sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java:26]: I guess this should be private, but also didn't see where it's used - is this gonna be in another PR?

File: sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java [hunk 4]
```diff
@@ -41,6 +41,20 @@ final class AndroidTransactionProfiler implements ITransactionProfiler {
   private long transactionStartNanos;
   private long profileStartCpuMillis;
 
+  public AndroidTransactionProfiler(
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java:44]: is this ctor needed?
I'm going to create an instance of the profiler without the `SentryAndroidOptions` (for startup profiling), so i'd like to avoid adding other constructors with options as params, if possible

File: sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java (hunk 1 above)
Comment (markushi) [reply sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java:80]: Yeah, that's on purpose. I didn't want to interfere with Object.clone() / the clonable interface here
image

File: sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java (hunk 3 above)
Comment (markushi) [reply sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java:26]: Oh true, we can remove this within this PR - thanks!

File: sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
+package io.sentry.android.core;
+
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+final class FrameMetrics {
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/FrameMetrics.java:6]: Can you rename it to `SentryFrameMetrics`, to distinguish it from the Android's one?

File: sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+      long durationNanos,
+      long delayNanos,
+      boolean isSlow,
+      boolean isFrozen,
+      float refreshRate) {
+
+    lastRefreshRate = (int) (refreshRate * 100) / 100F;
+
+    if (isFrozen) {
+      currentFrameMetrics.addFrozenFrame(durationNanos);
```
Comment (stefanosiano) [inline sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java:120]: should these be `delayNanos`? 
Otherwise you add the full frame duration as slow frame

Comment (markushi) [reply sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java]: True, we should collect the delay instead of the total duration 👍
